dev/com.ibm.ws.classloading/src/com/ibm/wsspi/classloading/ClassLoadingService.java

```diff
@@ -198,4 +199,10 @@
      * @param protectionDomainMap
      */
     void setSharedLibraryProtectionDomains(Map<String, ProtectionDomain> protectionDomainMap);
+
+    /**
+     * Returns a list of Libraries that are attached to application classloaders
+     * without configuration, for the purposes of providing application extensions
+     */
+    List<ApplicationExtensionLibrary> getAppExtLibs();
```

Comment:
Is this only used for apps that have no config? How do apps that have config get the libraries?

Also, this seems like the wrong place to document how the return value is used.

Reply:
This can potentially be used for any applications -- whether or not they have a nested `<classloader>` configured.  The idea is that:
```xml
<application location="myapp.war"/>
```
becomes:
```xml
<application location="myapp.war">
  <classloader privateLibraryRef="com.ibm.ws.jsfContainer.2.2"/>
</application>
```
and likewise this:
```xml
<application location="myapp.war">
  <classloader privateLibraryRef="alexandria"/>
</application>
```
becomes the following:
```xml
<application location="myapp.war">
  <classloader privateLibraryRef="alexandria,com.ibm.ws.jsfContainer.2.2"/>
</application>
```